pkg/kubelet/kubelet.go

```diff
@@ -1728,29 +1729,43 @@ func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, string) {
 	if ok, failReason, failMessage := kl.canAdmitPod(otherActivePods, pod); !ok {
 		// Log reason and return. Let the next sync iteration retry the resize
 		klog.V(2).Infof("Pod '%s' resize cannot be accommodated. Reason: '%s' Message: '%s'", pod.Name, failReason, failMessage)
-		return false, ""
+		return false, nil
 	}
 
-	var containersPatchData string
-	for _, container := range pod.Spec.Containers {
-		var resourcesPatchData string
-		for rName, rQuantity := range container.Resources.Requests {
-			container.ResourcesAllocated[rName] = rQuantity
-			resourcesPatchData += fmt.Sprintf(`"%s":"%s",`, rName, rQuantity.String())
+	podCopy := pod.DeepCopy()
+	for _, workload := range podCopy.Spec.Workloads() {
+		for rName, rQuantity := range workload.Resources.Requests {
```

Comment:
should this be the limits of the resources or both requests and limits ?

Reply:
only requests here. requests is the node capacity reservation done by kubelet. and kubelet always attempts to reconcile desired limits once apiserver has validated it and update the podspec.